datafusion/physical-plan/src/sorts/sort.rs

```diff
@@ -616,11 +617,7 @@ pub fn sort_batch(
         lexsort_to_indices(&sort_columns, fetch)?
     };
 
-    let columns = batch
-        .columns()
-        .iter()
-        .map(|c| take(c.as_ref(), &indices, None))
-        .collect::<Result<_, _>>()?;
+    let columns = get_arrayref_at_indices(batch.columns(), &indices)?;
```

Comment:
👍 

Maybe we should rename `get_arrayref_at_indices` to `take_arrays` or something 🤔  (as a follow on PR) to be more consistent

Reply:
This makes sense to me